src/classes/ERR_RecordError.cls

```diff
@@ -0,0 +1,536 @@
+
+        UTIL_Debug.debug('**** updateRecordError() returns error: ' + error);
+        return error;
+    }
+
+    /*******************************************************************************************************
+    * @description Updates an existing error record associated with a specific context and record from the result of a DML
+    * operation performed through the Database class.  Full_Message__c will be prepended with the latest message and timestamp, and
+    * truncated to 5K.  The Error Type will be updated based on the StatusCode of the Result.  The Datetime field will be updated
+    * to the current timestamp.  Retry_Pending__c will be set to true, indicated it should try the record again, if possible.
+    * @param error The error record to be updated.
+    * @param message The full message of all the errors for a given job and compiled IDs.
+    * @param statusCode The code associated with the error
+    * @return The error record.
+    */
+    @testVisible private static Error__c updateRecordError (Error__c error, String message, String statusCode) {
+        error.Datetime__c = System.now();
+        error.Retry_Pending__c = true;
+        error.Error_Type__c = statusCode;
+        error.Full_Message__c = updateFullMessage(error.Datetime__c, error.Full_Message__c, message);
+
+        UTIL_Debug.debug('**** updateRecordError() returns error: ' + error);
+        return error;
+    }
+
+    /*******************************************************************************************************
+    * @description For existing error records, it will concatenate the new message, and any existing text with a separator in between.
+    * @param ts The timestamp the associated
+    * @param context The context the exception was raised on.
+    * @param objectType The type of all records in the job, only to be passed in if they all match.
+    * @return The error record.
+    */
+    @testVisible private static String updateFullMessage(Datetime ts, String existingMessage, String message) {
+
+        String updatedMessage = ts.format() + ':\r\n' + message;
+
+        if (String.isNotBlank(existingMessage)) {
+            updatedMessage += '\r\n' + '========================================\r\n' + existingMessage;
+            // Limit to 5K worth of Message History
+            updatedMessage = updatedMessage.abbreviate(5120);
+        }
+
+        return updatedMessage;
+    }
+
+    /*******************************************************************************************************
+    * @description Returns the URL of a record, if available.
+    * @param recordId The id of the record associated with the error
+    * @return The URL of the record. Warning: it may not necessarily be valid.
+    */
+    private static String getRecordURL(String recordId) {
+        String recordURL = URL.getSalesforceBaseUrl().toExternalForm() + '/' + recordId;
```

Comment:
Isn't this mostly the same as the version in `ERR_Handler`? If so, I recommend updating that method (to clean it up) and using that version rather that duplicating.

Reply:
If you are ok with changing the visibility on the original, I will do so.